pandas/core/indexes/period.py

```diff
@@ -761,17 +761,19 @@ def _sub_datelike(self, other):
         return NotImplemented
 
     def _sub_period(self, other):
+        # If the operation is well-defined, we return an object-Index
+        # of DateOffsets.  Null entries are filled with pd.NaT
         if self.freq != other.freq:
             msg = _DIFFERENT_FREQ_INDEX.format(self.freqstr, other.freqstr)
             raise IncompatibleFrequency(msg)
 
         asi8 = self.asi8
         new_data = asi8 - other.ordinal
+        new_data = np.array([self.freq * x for x in new_data])
```

Comment:
It might be nice to implement DateOffset * arraylike so we don't have to right this broadcasting repeatedly.